src/ir/ty.rs

```diff
@@ -835,6 +835,22 @@ impl Type {
             _ => {}
         }
 
+        // Objective C template type parameter
+        if ty_kind == CXType_Typedef
```

Comment:
I don't think there is anything that guarantees that, and I believe this doesn't quite work with clang 4.0 anymore. But I'm kinda of lost how to address this.

The problem was that when there are multiple classes with the same template param name (ObjectType), and that ended up as multiple typedefs for the same name, at the root level.

The idea was, "oh skip the typedef, rewrite it as 'id' as thats valid and easy", but I'm still a bit green understanding how template params work in clang and bindgen.

Other idea was to get the "parent/module" level correct so that the type would be per trait, but again I was kind of lost on how I would make that happen.

I suppose it would be kind of neat to have an actual template support, but I don't think I have the stamina to even attempt that.